runtime/doc/news.txt

```diff
@@ -271,6 +271,10 @@ The following new APIs and features were added.
   • |vim.lsp.buf.definition()|, |vim.lsp.buf.declaration()|,
     |vim.lsp.buf.type_definition()|, and |vim.lsp.buf.implementation()| now
     support the `loclist` field of |vim.lsp.ListOpts|.
+  • Add `vim.Diagnostic.related_information` field and fill it with
+    `lsp.DiagnosticRelatedInformation[]`. Deprecate
```

Comment:
```suggestion
  • Add `vim.Diagnostic.related_information` field and fill it with
    `vim.diagnostic.RelatedInformation[]`. Deprecate
```

Reply:
I intentionally wrote `lsp.DiagnosticRelatedInformation` here though, to highlight the new logic I added in `lsp/diagnostic.lua`

Reply:
`lsp.DiagnosticRelatedInformation[]` does not link to anything in Nvim's help docs so I'm not sure it's valuable to have here in the user facing news. `vim.diagnostic.RelatedInformation` does have a help tag though. Maybe we keep this simple and just say:

```suggestion
• Add a new `related_information` field to |vim.Diagnostic| in favor of
  `vim.Diagnostic.user_data.lsp.relatedInformation` (now deprecated). See
  |vim.diagnostic.RelatedInformation|.
```

Reply:
sounds good, done